### ISSUE TITLE
FriendAccessAllowedAttribute public?

### ISSUE THREAD

**Issue by lukdr** [dotnet/coreclr#18984]
Would it be possible to make the System.Runtime.CompilerServices.FriendAccessAllowedAttribute public, to be more explicit and clearly state intent of what exactly should be visible to a friend assembly?

**Comment by jkotas** [comment]
https://github.com/dotnet/csharplang/issues/1724

This would need to be combined with enforcement in Roslyn or some other tool be actually useful.

**Comment by lukdr** [comment]
Let me also add that InternalsVisibleToAttribute has a `public bool AllInternalsVisible` property which is supposed to be used together with FriendAccessAllowedAttribute as mentioned in the summary:
```
///     If AllInternalsVisible is not true for a friend assembly, the FriendAccessAllowed attribute
///     indicates which internals are shared with that friend assembly.
```
This is not possible as the latter is internal.

**Comment by jkotas** [comment]
This comment is left-over from the times when we have experimented with this in .NET Framework. .NET Core does not use it - we did not like the complexity associated with it. 

The [official documentation]( 
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute.allinternalsvisible?view=netcore-2.1#System_Runtime_CompilerServices_InternalsVisibleToAttribute_AllInternalsVisible] just says that this is not implemented.
